app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -0,0 +1,490 @@
+        hibernateAccount.setEmail(genericAccount.getEmail());
+        hibernateAccount.setHealthCode(genericAccount.getHealthCode());
+        hibernateAccount.setHealthId(genericAccount.getHealthId());
+        hibernateAccount.setFirstName(genericAccount.getFirstName());
+        hibernateAccount.setLastName(genericAccount.getLastName());
+        hibernateAccount.setPasswordAlgorithm(genericAccount.getPasswordAlgorithm());
+        hibernateAccount.setPasswordHash(genericAccount.getPasswordHash());
+        hibernateAccount.setRoles(genericAccount.getRoles());
+        hibernateAccount.setStatus(genericAccount.getStatus());
+
+        // Attributes that need parsing.
+        if (genericAccount.getStudyIdentifier() != null) {
+            hibernateAccount.setStudyId(genericAccount.getStudyIdentifier().getIdentifier());
+        }
+        if (genericAccount.getCreatedOn() != null) {
+            hibernateAccount.setCreatedOn(genericAccount.getCreatedOn().getMillis());
+        }
+
+        // Attribute map
+        Map<String, String> hibernateAttrMap = hibernateAccount.getAttributes();
+        for (String oneAttrName : genericAccount.getAttributeNameSet()) {
+            hibernateAttrMap.put(oneAttrName, genericAccount.getAttribute(oneAttrName));
+        }
+
+        // Consents
+        Map<HibernateAccountConsentKey, HibernateAccountConsent> hibernateConsentMap = hibernateAccount.getConsents();
+        for (Map.Entry<SubpopulationGuid, List<ConsentSignature>> consentListForSubpop :
+                genericAccount.getAllConsentSignatureHistories().entrySet()) {
+            String subpopGuidString = consentListForSubpop.getKey().getGuid();
+
+            for (ConsentSignature oneConsent : consentListForSubpop.getValue()) {
+                // Consent key
+                HibernateAccountConsentKey hibernateConsentKey = new HibernateAccountConsentKey(subpopGuidString,
+                        oneConsent.getSignedOn());
+
+                // Simple consent attributes.
+                HibernateAccountConsent hibernateConsentValue = new HibernateAccountConsent();
+                hibernateConsentValue.setConsentCreatedOn(oneConsent.getConsentCreatedOn());
+                hibernateConsentValue.setName(oneConsent.getName());
+                hibernateConsentValue.setSignatureImageData(oneConsent.getImageData());
+                hibernateConsentValue.setSignatureImageMimeType(oneConsent.getImageMimeType());
+                hibernateConsentValue.setWithdrewOn(oneConsent.getWithdrewOn());
+
+                // We need to parse birthdate.
+                if (StringUtils.isNotBlank(oneConsent.getBirthdate())) {
+                    hibernateConsentValue.setBirthdate(oneConsent.getBirthdate());
+                }
+
+                // Store in hibernate account.
+                hibernateConsentMap.put(hibernateConsentKey, hibernateConsentValue);
+            }
+        }
+
+        return hibernateAccount;
+    }
```

Comment:
There are some very slight differences between HibernateAccount and GenericAccount, most notably around collections. For example, Stormpath doesn’t let us have the attribute map directly, while Hibernate does everything with that map directly. Also,the Account interface has a Map of Strings to a List of ConsentSignatures while Hibernate forces us to use… something else.

After we get rid of Stormpath, we could take steps to merge these together. But that largely depends on how tightly coupled you want to be to the Hibernate implementation. The consents stuff is super wonky, so I'm not sure if we want to use that in our interface.